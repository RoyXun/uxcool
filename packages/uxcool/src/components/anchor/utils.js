const REG_ANCHOR_ID = /#[^#]+$/;

export function getAnchorRelateElement(href = '') {
  const matches = href.match(REG_ANCHOR_ID);
  if (!matches) {
    return null;
  }

  return document.querySelector(matches[0]);
}

export function getElementOffsetTop(element, container) {
  let top = 0;
  if (!element || !container) {
    return top;
  }

  const box = element.getBoundingClientRect();
  if (container.window === window) {
    const doc = element.ownerDocument;
    const docElement = doc && doc.documentElement;
    const body = doc && doc.body;
    top = box.top - docElement.clientTop || body.clientTop || 0;
  } else {
    const containerRect = container.getBoundingClientRect();
    top = box.top - containerRect.top;
  }
  return top;
}
