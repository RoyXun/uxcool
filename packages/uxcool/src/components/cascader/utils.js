import { isFunction } from '@suning/v-utils';

export const SPLIT_STR = '\u0001';

export const DEFAULT_FIELD_NAMES = {
  label: 'label',
  value: 'value',
  children: 'children',
};

export function genCascaderKey(valuePath) {
  return valuePath.join(SPLIT_STR);
}

export function delay(fn, wait) {
  if (!isFunction(fn)) {
    return null;
  }
  return setTimeout(fn, wait);
}
