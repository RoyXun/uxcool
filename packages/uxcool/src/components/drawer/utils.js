import { isNaN, isNumber } from '@cloud-sn/v-utils';

export function genUid() {
  let dt = new Date().getTime();
  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
    // eslint-disable-next-line no-bitwise, no-mixed-operators
    const r = ((dt + Math.random() * 16) % 16) | 0;
    dt = Math.floor(dt / 16);
    // eslint-disable-next-line no-bitwise,
    return (c === 'x' ? r : (r & 0x3) | 0x8).toString(16);
  });
}

export function noop() {}

export function isNumeric(value) {
  return !isNaN(value) && isNumber(value);
}

export function hasScrollBar() {
  return document.body.clientWidth < window.innerWidth;
}

export function getVisibleDrawerCnt(drawers = {}) {
  return Object.keys(drawers).filter((k) => drawers[k]).length;
}

export function hasVisibleDrawer(drawers = {}) {
  return getVisibleDrawerCnt(drawers) >= 1;
}

export function getDoc() {
  return window ? window.document.body : {};
}
