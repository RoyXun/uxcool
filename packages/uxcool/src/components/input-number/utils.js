import isNaN from 'lodash/isNaN';

export { isNaN };

export function isNotCompleteNumber(v) {
  return (
    isNaN(Number(v)) ||
    v === '' ||
    v === null ||
    v === undefined ||
    (v && v.toString().slice('.') === v.length - 1)
  );
}

export function getPrecision(val, defaultPrecision = null) {
  if (defaultPrecision !== null) {
    return defaultPrecision;
  }

  let valStr = val.toString();
  let eDigit = 0;
  if (/e-/.test(valStr)) {
    [valStr, eDigit] = valStr.split('e-');
  }

  let decimal = 0;
  if (/\./.test(valStr)) {
    decimal = parseInt(valStr.split('.')[1].length, 10);
  }

  return decimal + eDigit;
}

export function getMaxPrecision(val, step, ratio = 1, defaultPrecision = null) {
  if (defaultPrecision !== null) {
    return defaultPrecision;
  }

  const stepPrecision = getPrecision(step);
  const ratioPrecision = getPrecision(ratio);
  const valPrecision = getPrecision(val);

  return Math.max(valPrecision, stepPrecision + ratioPrecision);
}

export function getRatio(e) {
  let ratio = 1;
  if (e.metaKey || e.ctrlKey) {
    ratio = 0.1;
  } else if (e.shiftKey) {
    ratio = 10;
  }
  return ratio;
}
