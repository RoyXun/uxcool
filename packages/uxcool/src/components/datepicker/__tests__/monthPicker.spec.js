import {
  mount,
  waitTime,
  getPortal,
  triggerEvent,
  mountPickerFactory,
  createWrapper,
} from '@suning/v-test-utils';
import dayjs from 'dayjs';

import { selectMonth } from './utils';
import MonthPicker from '../monthPicker';
import localeCN from '../locale/zh_CN';
import localeEN from '../locale/en_US';

describe('MonthPicker', () => {
  const mountPicker = mountPickerFactory(MonthPicker);
  const mockDate = dayjs('2019-05-03');
  const format = 'YYYY-MM';
  it('create', async () => {
    const wrapper = await mountPicker({
      propsData: {
        value: mockDate.toDate(),
        format,
        transitionName: '',
      },
    });
    await triggerEvent(wrapper, 'click');
    const portal = await getPortal(wrapper);

    await selectMonth(portal, mockDate.add(5, 'month'));

    expect(wrapper.find('.ux-input').element.value).toBe(mockDate.add(5, 'month').format(format));
  });
  it('disabled', async () => {
    const wrapper = await mountPicker({
      propsData: {
        disabled: true,
      },
    });

    expect(wrapper.find('.ux-input-disabled').exists()).toBeTruthy();
    await triggerEvent(wrapper, 'click');

    const portal = await getPortal(wrapper);
    expect(portal.element).toBeUndefined();
  });

  it('format', async () => {
    const customFormat = 'MM/YY';
    const wrapper = await mountPicker({
      propsData: {
        value: mockDate.toDate(),
        format: customFormat,
        transitionName: '',
      },
    });

    const input = wrapper.find('.ux-input');
    expect(input.element.value).toBe(mockDate.format(customFormat));
    await triggerEvent(wrapper, 'click');
    const portal = await getPortal(wrapper);

    await selectMonth(portal, mockDate.add(3, 'month'));
    expect(input.element.value).toBe(mockDate.add(3, 'month').format(customFormat));
  });

  it('get-popup-container', async () => {
    const Test = {
      render() {
        const { $refs } = this;
        const props = {
          getPopupContainer() {
            return $refs.wrapRef;
          },
        };
        return (
          <div ref="wrapRef">
            <MonthPicker {...{ props }} />
          </div>
        );
      },
    };
    const wrapper = mount(Test);

    await waitTime();
    await triggerEvent(wrapper.find(MonthPicker), 'click');

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('placement', async () => {
    const wrapper = await mountPicker({
      propsData: {
        transitionName: '',
      },
    });

    await triggerEvent(wrapper, 'click');

    const portal = await getPortal(wrapper);

    expect(portal.find('.ux-calendar-picker-container-placement-bottomLeft').exists()).toBeTruthy();

    wrapper.setProps({
      placement: 'topRight',
    });
    await waitTime();
    expect(portal.find('.ux-calendar-picker-container-placement-topRight').exists()).toBeTruthy();
  });

  it('placeholder', async () => {
    const wrapper = await mountPicker({
      propsData: {
        transitionName: '',
      },
    });

    const input = wrapper.find('.ux-input');
    expect(input.attributes('placeholder')).toBe(localeCN.lang.monthPlaceholder);

    const placeholder = 'this is month placholder';
    wrapper.setProps({
      placeholder,
    });
    await waitTime();

    expect(input.attributes('placeholder')).toBe(placeholder);
  });

  describe('allow-clear', () => {
    it('should clear when allow-clear is true', async () => {
      const wrapper = await mountPicker({
        propsData: {
          value: mockDate.toDate(),
          allowClear: true,
          format,
          transitionName: '',
        },
      });
      await triggerEvent(wrapper, 'click');
      const portal = await getPortal(wrapper);

      await selectMonth(portal, mockDate.add(1, 'month'));
      const input = wrapper.find('.ux-input');
      expect(input.element.value).toBe(mockDate.add(1, 'month').format(format));

      const clear = wrapper.find('.ux-calendar-picker-clear');
      expect(clear.exists()).toBeTruthy();

      await triggerEvent(clear, 'click');
      expect(input.element.value).toBe('');
    });
    it('should not clear when allow-clear is false', async () => {
      const wrapper = await mountPicker({
        propsData: {
          value: mockDate.toDate(),
          allowClear: false,
          transitionName: '',
        },
      });

      await triggerEvent(wrapper, 'click');

      const portal = await getPortal(wrapper);
      await selectMonth(portal, mockDate.add(3, 'month'));

      const input = wrapper.find('.ux-input');
      expect(input.element.value).toBe(mockDate.add(3, 'month').format(format));

      expect(wrapper.find('.ux-calendar-picker-clear').exists()).toBeFalsy();
    });
  });

  describe('size', () => {
    it('small', async () => {
      const wrapper = await mountPicker({
        propsData: {
          size: 'small',
          transitionName: '',
        },
      });

      expect(`${wrapper.html()}`).toMatchSnapshot();
    });
    it('default', async () => {
      const wrapper = await mountPicker({
        propsData: {
          size: 'default',
          transitionName: '',
        },
      });

      expect(`${wrapper.html()}`).toMatchSnapshot();
    });
    it('large', async () => {
      const wrapper = await mountPicker({
        propsData: {
          size: 'large',
          transitionName: '',
        },
      });
      expect(`${wrapper.html()}`).toMatchSnapshot();
    });
  });

  describe('visible', () => {
    it('should show trigger when visible is true', async () => {
      const wrapper = await mountPicker({
        propsData: {
          visible: true,
          transitionName: '',
        },
      });

      const portal = await getPortal(wrapper);
      expect(portal.find('.ux-calendar-picker-container').exists()).toBeTruthy();
    });
  });

  describe('value', () => {
    it('should update value after selected month', async () => {
      const onInput = jest.fn();
      const selectedMonth = mockDate.add(3, 'month');
      const wrapper = await mountPicker({
        propsData: {
          value: mockDate.toDate(),
          format,
          transitionName: '',
        },
        listeners: {
          input: onInput,
        },
      });

      await triggerEvent(wrapper, 'click');

      const portal = await getPortal(wrapper);
      await selectMonth(portal, selectedMonth);

      const input = wrapper.find('.ux-input');
      expect(input.element.value).toBe(selectedMonth.format(format));

      expect(onInput).toHaveBeenCalled();
      expect(onInput).toHaveBeenCalledWith(selectedMonth.toDate());
    });

    it('should select month when init value', async () => {
      const onInput = jest.fn();
      const selectedDate = mockDate.add(5, 'month');
      const wrapper = await mountPicker({
        propsData: {
          value: selectedDate.toDate(),
          format,
          transitionName: '',
        },
      });

      await triggerEvent(wrapper, 'click');
      const portal = await getPortal(wrapper);

      expect(wrapper.find('.ux-input').element.value).toBe(selectedDate.format(format));
      expect(portal.find('.ux-calendar-month-panel-year-select-content').text()).toBe(
        selectedDate.format('YYYY')
      );
      expect(portal.find('.ux-calendar-month-panel-selected-cell').attributes('title')).toBe(
        selectedDate.format('M月')
      );

      expect(onInput).not.toHaveBeenCalled();
    });

    it('should select month after update value', async () => {
      const onInput = jest.fn();
      const updateValue = mockDate.add(6, 'month');
      const wrapper = await mountPicker({
        propsData: {
          format,
          trnasitionName: '',
        },
        listeners: {
          input: onInput,
        },
      });

      await triggerEvent(wrapper, 'click');
      wrapper.setProps({
        value: updateValue.toDate(),
      });

      await waitTime();

      const portal = await getPortal(wrapper);

      expect(wrapper.find('.ux-input').element.value).toBe(updateValue.format(format));
      expect(portal.find('.ux-calendar-month-panel-year-select-content').text()).toBe(
        updateValue.format('YYYY')
      );
      expect(portal.find('.ux-calendar-month-panel-selected-cell').attributes('title')).toBe(
        updateValue.format('M月')
      );
      expect(onInput).not.toHaveBeenCalled();
    });
  });

  describe('locale', () => {
    it('should work when use locale.lang', async () => {
      const wrapper = await mountPicker({
        propsData: {
          value: mockDate.toDate(),
          locale: localeEN.lang,
          trnasitionName: '',
        },
      });
      await triggerEvent(wrapper, 'click');
      const portal = await getPortal(wrapper);
      expect(`${wrapper.html()}${portal.html()}`).toMatchSnapshot();
    });
    it('should work when use locale', async () => {
      const wrapper = await mountPicker({
        propsData: {
          value: mockDate.toDate(),
          locale: localeEN,
          trnasitionName: '',
        },
      });
      await triggerEvent(wrapper, 'click');
      const portal = await getPortal(wrapper);
      expect(`${wrapper.html()}${portal.html()}`).toMatchSnapshot();
    });
  });

  describe('disabled-month', () => {
    it('should disabled 2月,5月,7月, 8月, 12月', async () => {
      function disabledMonth(date) {
        const m = dayjs(date).month();
        return (
          m
          in {
            1: 1,
            4: 1,
            6: 1,
            7: 1,
            11: 1,
          }
        );
      }
      const wrapper = await mountPicker({
        propsData: {
          value: mockDate.toDate(),
          disabledMonth,
          trnasitionName: '',
        },
      });

      await triggerEvent(wrapper, 'click');
      const portal = await getPortal(wrapper);

      expect(portal.html()).toMatchSnapshot();
    });
  });

  describe('events', () => {
    it('change', async () => {
      const onChange = jest.fn();
      const selectedValue = mockDate.add(2, 'month');
      const wrapper = await mountPicker({
        propsData: {
          value: mockDate.toDate(),
          format,
          trnasitionName: '',
        },
        listeners: {
          change: onChange,
        },
      });

      await triggerEvent(wrapper, 'click');

      const portal = await getPortal(wrapper);

      await selectMonth(portal, selectedValue);

      expect(onChange).toHaveBeenCalled();
      expect(onChange.mock.calls[0]).toEqual([
        selectedValue.toDate(),
        selectedValue.format(format),
      ]);
    });
    it('open-change', async () => {
      const onOpenChange = jest.fn();
      const selectedValue = mockDate.add(2, 'month');
      const wrapper = await mountPicker({
        propsData: {
          trnasitionName: '',
        },
        listeners: {
          'open-change': onOpenChange,
        },
      });

      await triggerEvent(wrapper, 'click');
      expect(onOpenChange.mock.calls.length).toBe(1);
      expect(onOpenChange.mock.calls[0]).toEqual([true]);

      const portal = await getPortal(wrapper);
      await selectMonth(portal, selectedValue);

      expect(onOpenChange.mock.calls.length).toBe(2);
      expect(onOpenChange.mock.calls[1]).toEqual([false]);

      await triggerEvent(wrapper, 'click');
      expect(onOpenChange.mock.calls.length).toBe(3);
      expect(onOpenChange.mock.calls[2]).toEqual([true]);
      const body = createWrapper(wrapper.element.ownerDocument.body);

      await triggerEvent(body, 'mousedown');
      expect(onOpenChange.mock.calls.length).toBe(4);
      expect(onOpenChange.mock.calls[3]).toEqual([false]);
    });

    it('panel-change', async () => {
      const onPanelChange = jest.fn();
      const wrapper = await mountPicker({
        propsData: {
          trnasitionName: '',
        },
        listeners: {
          'panel-change': onPanelChange,
        },
      });

      await triggerEvent(wrapper, 'click');
      const portal = await getPortal(wrapper);

      await triggerEvent(portal.find('.ux-calendar-month-panel-year-select'), 'click');

      expect(onPanelChange.mock.calls.length).toBe(1);
      expect(onPanelChange.mock.calls[0][1]).toBe('year');

      await triggerEvent(portal.find('.ux-calendar-year-panel-decade-select'), 'click');
      expect(onPanelChange.mock.calls.length).toBe(2);
      expect(onPanelChange.mock.calls[1][1]).toBe('decade');

      await triggerEvent(portal.findAll('.ux-calendar-decade-panel-cell').at(1), 'click');
      expect(onPanelChange.mock.calls.length).toBe(3);
      expect(onPanelChange.mock.calls[2][1]).toBe('year');

      await triggerEvent(portal.findAll('.ux-calendar-year-panel-cell').at(1), 'click');
      expect(onPanelChange.mock.calls.length).toBe(4);
      expect(onPanelChange.mock.calls[3][1]).toBe('month');
    });
  });
});
