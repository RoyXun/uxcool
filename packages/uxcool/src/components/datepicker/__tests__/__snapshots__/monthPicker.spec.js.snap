// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthPicker disabled-month should disabled 2月,5月,7月, 8月, 12月 1`] = `
<div role="popup">
  <div role="align-popup" class="ux-calendar-picker-container ux-calendar-picker-container-placement-bottomLeft slide-up-enter slide-up-enter-active" style="left: -999px; top: -1002px;">
    <div class="ux-calendar-picker-container-content">
      <div class="ux-calendar ux-calendar-month ux-calendar-month-calendar">
        <div class="ux-calendar-month-calendar-content">
          <div class="ux-calendar-month-header-wrap">
            <div class="ux-calendar-header">
              <div style="position: relative;"><a title="上一年 (Control键加左方向键)" role="button" class="ux-calendar-prev-year-btn"></a> <a title="上个月 (翻页上键)" role="button" class="ux-calendar-prev-month-btn"></a> <span class="ux-calendar-ym-select"><a role="button" title="选择年份" class="ux-calendar-year-select">2019年</a><a role="button" title="选择月份" class="ux-calendar-month-select">5月</a></span> <a title="下个月 (翻页下键)" role="button" class="ux-calendar-next-month-btn"></a> <a title="下一年 (Control键加右方向键)" role="button" class="ux-calendar-next-year-btn"></a></div>
              <div>
                <div class="ux-calendar-month-panel">
                  <div class="ux-calendar-month-panel-header"><a title="上一年 (Control键加左方向键)" role="button" class="ux-calendar-month-panel-prev-year-btn"></a> <a title="选择年份" role="button" class="ux-calendar-month-panel-year-select"><span class="ux-calendar-month-panel-year-select-content">
        2019
      </span></a> <a title="下一年 (Control键加右方向键)" role="button" class="ux-calendar-month-panel-next-year-btn"></a></div>
                  <div class="ux-calendar-month-panel-body">
                    <table role="monthgrid" cellspacing="0" class="ux-calendar-month-panel-table">
                      <tbody class="ux-calendar-month-panel-tbody">
                        <tr role="row">
                          <td role="gridcell" title="1月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">1月</a></td>
                          <td role="gridcell" title="2月" class="ux-calendar-month-panel-cell ux-calendar-month-panel-cell-disabled"><a class="ux-calendar-month-panel-month">2月</a></td>
                          <td role="gridcell" title="3月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">3月</a></td>
                        </tr>
                        <tr role="row">
                          <td role="gridcell" title="4月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">4月</a></td>
                          <td role="gridcell" title="5月" class="ux-calendar-month-panel-cell ux-calendar-month-panel-cell-disabled ux-calendar-month-panel-selected-cell"><a class="ux-calendar-month-panel-month">5月</a></td>
                          <td role="gridcell" title="6月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">6月</a></td>
                        </tr>
                        <tr role="row">
                          <td role="gridcell" title="7月" class="ux-calendar-month-panel-cell ux-calendar-month-panel-cell-disabled"><a class="ux-calendar-month-panel-month">7月</a></td>
                          <td role="gridcell" title="8月" class="ux-calendar-month-panel-cell ux-calendar-month-panel-cell-disabled"><a class="ux-calendar-month-panel-month">8月</a></td>
                          <td role="gridcell" title="9月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">9月</a></td>
                        </tr>
                        <tr role="row">
                          <td role="gridcell" title="10月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">10月</a></td>
                          <td role="gridcell" title="11月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">11月</a></td>
                          <td role="gridcell" title="12月" class="ux-calendar-month-panel-cell ux-calendar-month-panel-cell-disabled"><a class="ux-calendar-month-panel-month">12月</a></td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
                <!---->
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MonthPicker get-popup-container 1`] = `
<div>
  <div class="ux-calendar-picker ux-calendar-picker-default" style=""><input type="text" readonly="readonly" placeholder="请选择月份" class="ux-input ux-calendar-picker-input"><span class="ux-calendar-picker-icon"></span></div>
  <div role="popup">
    <div role="align-popup" class="ux-calendar-picker-container ux-calendar-picker-container-placement-bottomLeft slide-up-enter slide-up-enter-active" style="left: -999px; top: -1002px;">
      <div class="ux-calendar-picker-container-content">
        <div class="ux-calendar ux-calendar-month ux-calendar-month-calendar">
          <div class="ux-calendar-month-calendar-content">
            <div class="ux-calendar-month-header-wrap">
              <div class="ux-calendar-header">
                <div style="position: relative;"><a title="上一年 (Control键加左方向键)" role="button" class="ux-calendar-prev-year-btn"></a> <a title="上个月 (翻页上键)" role="button" class="ux-calendar-prev-month-btn"></a> <span class="ux-calendar-ym-select"><a role="button" title="选择年份" class="ux-calendar-year-select">2020年</a><a role="button" title="选择月份" class="ux-calendar-month-select">1月</a></span> <a title="下个月 (翻页下键)" role="button" class="ux-calendar-next-month-btn"></a> <a title="下一年 (Control键加右方向键)" role="button" class="ux-calendar-next-year-btn"></a></div>
                <div>
                  <div class="ux-calendar-month-panel">
                    <div class="ux-calendar-month-panel-header"><a title="上一年 (Control键加左方向键)" role="button" class="ux-calendar-month-panel-prev-year-btn"></a> <a title="选择年份" role="button" class="ux-calendar-month-panel-year-select"><span class="ux-calendar-month-panel-year-select-content">
        2020
      </span></a> <a title="下一年 (Control键加右方向键)" role="button" class="ux-calendar-month-panel-next-year-btn"></a></div>
                    <div class="ux-calendar-month-panel-body">
                      <table role="monthgrid" cellspacing="0" class="ux-calendar-month-panel-table">
                        <tbody class="ux-calendar-month-panel-tbody">
                          <tr role="row">
                            <td role="gridcell" title="1月" class="ux-calendar-month-panel-cell ux-calendar-month-panel-selected-cell ux-calendar-month-panel-current-cell"><a class="ux-calendar-month-panel-month">1月</a></td>
                            <td role="gridcell" title="2月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">2月</a></td>
                            <td role="gridcell" title="3月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">3月</a></td>
                          </tr>
                          <tr role="row">
                            <td role="gridcell" title="4月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">4月</a></td>
                            <td role="gridcell" title="5月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">5月</a></td>
                            <td role="gridcell" title="6月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">6月</a></td>
                          </tr>
                          <tr role="row">
                            <td role="gridcell" title="7月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">7月</a></td>
                            <td role="gridcell" title="8月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">8月</a></td>
                            <td role="gridcell" title="9月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">9月</a></td>
                          </tr>
                          <tr role="row">
                            <td role="gridcell" title="10月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">10月</a></td>
                            <td role="gridcell" title="11月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">11月</a></td>
                            <td role="gridcell" title="12月" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">12月</a></td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                  <!---->
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MonthPicker locale should work when use locale 1`] = `
<div class="ux-calendar-picker ux-calendar-picker-default" trnasitionname="" style=""><input type="text" readonly="readonly" placeholder="Select month" class="ux-input ux-calendar-picker-input"><i class="fu fu-close_circle ux-calendar-picker-clear"></i><span class="ux-calendar-picker-icon"></span></div>
<div role="popup">
  <div role="align-popup" class="ux-calendar-picker-container ux-calendar-picker-container-placement-bottomLeft slide-up-enter slide-up-enter-active" style="left: -999px; top: -1002px;">
    <div class="ux-calendar-picker-container-content">
      <div class="ux-calendar ux-calendar-month ux-calendar-month-calendar">
        <div class="ux-calendar-month-calendar-content">
          <div class="ux-calendar-month-header-wrap">
            <div class="ux-calendar-header">
              <div style="position: relative;"><a title="Last year (Control + left)" role="button" class="ux-calendar-prev-year-btn"></a> <a title="Previous month (PageUp)" role="button" class="ux-calendar-prev-month-btn"></a> <span class="ux-calendar-my-select"><a role="button" title="Choose a month" class="ux-calendar-month-select">May</a><a role="button" title="Choose a year" class="ux-calendar-year-select">2019</a></span> <a title="Next month (PageDown)" role="button" class="ux-calendar-next-month-btn"></a> <a title="Next year (Control + right)" role="button" class="ux-calendar-next-year-btn"></a></div>
              <div>
                <div class="ux-calendar-month-panel">
                  <div class="ux-calendar-month-panel-header"><a title="Last year (Control + left)" role="button" class="ux-calendar-month-panel-prev-year-btn"></a> <a title="Choose a year" role="button" class="ux-calendar-month-panel-year-select"><span class="ux-calendar-month-panel-year-select-content">
        2019
      </span></a> <a title="Next year (Control + right)" role="button" class="ux-calendar-month-panel-next-year-btn"></a></div>
                  <div class="ux-calendar-month-panel-body">
                    <table role="monthgrid" cellspacing="0" class="ux-calendar-month-panel-table">
                      <tbody class="ux-calendar-month-panel-tbody">
                        <tr role="row">
                          <td role="gridcell" title="Jan" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Jan</a></td>
                          <td role="gridcell" title="Feb" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Feb</a></td>
                          <td role="gridcell" title="Mar" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Mar</a></td>
                        </tr>
                        <tr role="row">
                          <td role="gridcell" title="Apr" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Apr</a></td>
                          <td role="gridcell" title="May" class="ux-calendar-month-panel-cell ux-calendar-month-panel-selected-cell"><a class="ux-calendar-month-panel-month">May</a></td>
                          <td role="gridcell" title="Jun" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Jun</a></td>
                        </tr>
                        <tr role="row">
                          <td role="gridcell" title="Jul" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Jul</a></td>
                          <td role="gridcell" title="Aug" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Aug</a></td>
                          <td role="gridcell" title="Sep" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Sep</a></td>
                        </tr>
                        <tr role="row">
                          <td role="gridcell" title="Oct" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Oct</a></td>
                          <td role="gridcell" title="Nov" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Nov</a></td>
                          <td role="gridcell" title="Dec" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Dec</a></td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
                <!---->
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MonthPicker locale should work when use locale.lang 1`] = `
<div class="ux-calendar-picker ux-calendar-picker-default" trnasitionname="" style=""><input type="text" readonly="readonly" placeholder="Select month" class="ux-input ux-calendar-picker-input"><i class="fu fu-close_circle ux-calendar-picker-clear"></i><span class="ux-calendar-picker-icon"></span></div>
<div role="popup">
  <div role="align-popup" class="ux-calendar-picker-container ux-calendar-picker-container-placement-bottomLeft slide-up-enter slide-up-enter-active" style="left: -999px; top: -1002px;">
    <div class="ux-calendar-picker-container-content">
      <div class="ux-calendar ux-calendar-month ux-calendar-month-calendar">
        <div class="ux-calendar-month-calendar-content">
          <div class="ux-calendar-month-header-wrap">
            <div class="ux-calendar-header">
              <div style="position: relative;"><a title="Last year (Control + left)" role="button" class="ux-calendar-prev-year-btn"></a> <a title="Previous month (PageUp)" role="button" class="ux-calendar-prev-month-btn"></a> <span class="ux-calendar-my-select"><a role="button" title="Choose a month" class="ux-calendar-month-select">May</a><a role="button" title="Choose a year" class="ux-calendar-year-select">2019</a></span> <a title="Next month (PageDown)" role="button" class="ux-calendar-next-month-btn"></a> <a title="Next year (Control + right)" role="button" class="ux-calendar-next-year-btn"></a></div>
              <div>
                <div class="ux-calendar-month-panel">
                  <div class="ux-calendar-month-panel-header"><a title="Last year (Control + left)" role="button" class="ux-calendar-month-panel-prev-year-btn"></a> <a title="Choose a year" role="button" class="ux-calendar-month-panel-year-select"><span class="ux-calendar-month-panel-year-select-content">
        2019
      </span></a> <a title="Next year (Control + right)" role="button" class="ux-calendar-month-panel-next-year-btn"></a></div>
                  <div class="ux-calendar-month-panel-body">
                    <table role="monthgrid" cellspacing="0" class="ux-calendar-month-panel-table">
                      <tbody class="ux-calendar-month-panel-tbody">
                        <tr role="row">
                          <td role="gridcell" title="Jan" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Jan</a></td>
                          <td role="gridcell" title="Feb" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Feb</a></td>
                          <td role="gridcell" title="Mar" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Mar</a></td>
                        </tr>
                        <tr role="row">
                          <td role="gridcell" title="Apr" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Apr</a></td>
                          <td role="gridcell" title="May" class="ux-calendar-month-panel-cell ux-calendar-month-panel-selected-cell"><a class="ux-calendar-month-panel-month">May</a></td>
                          <td role="gridcell" title="Jun" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Jun</a></td>
                        </tr>
                        <tr role="row">
                          <td role="gridcell" title="Jul" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Jul</a></td>
                          <td role="gridcell" title="Aug" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Aug</a></td>
                          <td role="gridcell" title="Sep" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Sep</a></td>
                        </tr>
                        <tr role="row">
                          <td role="gridcell" title="Oct" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Oct</a></td>
                          <td role="gridcell" title="Nov" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Nov</a></td>
                          <td role="gridcell" title="Dec" class="ux-calendar-month-panel-cell"><a class="ux-calendar-month-panel-month">Dec</a></td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
                <!---->
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MonthPicker size default 1`] = `<div class="ux-calendar-picker ux-calendar-picker-default"><input type="text" readonly="readonly" placeholder="请选择月份" class="ux-input ux-calendar-picker-input"><span class="ux-calendar-picker-icon"></span></div>`;

exports[`MonthPicker size large 1`] = `<div class="ux-calendar-picker ux-calendar-picker-large"><input type="text" readonly="readonly" placeholder="请选择月份" class="ux-input ux-calendar-picker-input ux-input-lg"><span class="ux-calendar-picker-icon"></span></div>`;

exports[`MonthPicker size small 1`] = `<div class="ux-calendar-picker ux-calendar-picker-small"><input type="text" readonly="readonly" placeholder="请选择月份" class="ux-input ux-calendar-picker-input ux-input-sm"><span class="ux-calendar-picker-icon"></span></div>`;
